jena-arq/Grammar/master.jj

```diff
@@ -2166,6 +2227,17 @@ String String() : { Token t ; String lex ; }
     }
 }
 
+Number Number() : { Token t ; Number number ; }
+{
```

Comment:
Number -> JSONNumber

But will rule NumericLiteral work here? Then parse the lexical form to a number.

However, generally, working in nodes is probably going to be easier because bindings map variables to Nodes.